lib/internal/errors.js

```diff
@@ -145,28 +147,21 @@ function makeSystemErrorWithCode(key) {
   };
 }
 
-let useOriginalName = false;
-
 function makeNodeErrorWithCode(Base, key) {
+  // TODO(BridgeAR): Use a single `NodeError` class and expose that to the
+  // users. That way it's possible to match against those.
```

Comment:
> That way it's possible to match against those.

What's going to be matched? `toString()`? Why would it require exposing `NodeError`?
I don't think it's really a great idea to expose internals like this. For checking the hierarchy, something like `utils.isNodeError()` would already suffice, or we could expose other individual methods instead.

Reply:
I meant an `instanceof` check against the class. You are right that `util.isNodeError` would be better here to prevent any user code to create such errors and we could make it work cross realms.
I'll update the comment.

Reply:
I decided to remove the comment. On a very long term perspective I'd like to have a way to have something like in Java:

```js
try {
  // foobar
} catch NodeError (e) {
  // This catches instances of NodeError (the class would have to be defined in the scope).
} catch (e) {
  // All other errors.
}
```

This would require the a dramatic language change which I do not see right now and on a short term it's likely good to expose the utility function instead.